components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -180,55 +200,201 @@ export default function UsageBasedBillingConfig({ subject, attributionId }: Prop
-                        <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Billing</div>
+                    <div className="flex flex-col mt-4 h-52 p-4 rounded-xl bg-gray-50 dark:bg-gray-900">
+                        <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Balance</div>
                         <Spinner className="m-2 h-5 w-5 animate-spin" />
                     </div>
                 )}
-                {showUpgradeBilling && (
-                    <div className="flex flex-col mt-4 h-32 p-4 rounded-xl bg-gray-100 dark:bg-gray-800">
-                        <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Billing</div>
-                        <div className="text-xl font-semibold flex-grow text-gray-600 dark:text-gray-400">Inactive</div>
-                        <button className="self-end" onClick={() => setShowBillingSetupModal(true)}>
-                            Upgrade Billing
+                {showBalance && (
+                    <div className="flex flex-col mt-4 p-4 rounded-xl bg-gray-50 dark:bg-gray-900">
+                        <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Balance</div>
+                        <div className="mt-1 text-xl font-semibold flex-grow">
+                            <span className="text-gray-900 dark:text-gray-100">
+                                {typeof currentUsage === "number" ? Math.round(currentUsage) : "?"}
+                            </span>
+                            <span className="text-gray-400 dark:text-gray-500">
+                                {" "}
+                                / {usageLimit} Credit{usageLimit === 1 ? "" : "s"}
+                            </span>
+                        </div>
+                        <div className="mt-4 text-sm flex">
+                            <span className="flex-grow">
+                                {showManageBilling && (
+                                    <button className="gp-link" onClick={() => setShowUpdateLimitModal(true)}>
+                                        Manage Usage Limit
+                                    </button>
+                                )}
+                            </span>
+                            {typeof currentUsage === "number" && typeof usageLimit === "number" && (
+                                <span className="text-gray-400 dark:text-gray-500">
+                                    {Math.round((100 * currentUsage) / usageLimit)}% used
+                                </span>
+                            )}
+                        </div>
+                        <div className="mt-2 flex">
+                            <progress className="h-4 flex-grow rounded-xl" value={currentUsage} max={usageLimit} />
+                        </div>
+                        <div className="bg-gray-100 dark:bg-gray-800 border-t border-gray-200 dark:border-gray-800 -m-4 p-4 mt-4 rounded-b-xl flex">
```

Comment:
Thanks for suggesting better colors! However, in order to make implementation easy, I would love us to stick to a consistent mapping between light and dark colors.

Currently in our code base,
- with `bg-gray-100`, we use mostly `dark:bg-gray-800`, but occasionally also `dark:bg-gray-700` and `dark:bg-gray-900` -- it would be nice to make all of these consistent if possible
- with `border-gray-200`, we use overwhelmingly `dark:border-gray-800`, but also `dark:border-gray-700` and `dark:border-gray-600` once or twice

| With `bg-gray-100` | With `border-gray-200` |
| --- | --- |
| <img width="559" alt="Screenshot 2022-10-07 at 11 00 56" src="https://user-images.githubusercontent.com/599268/194516657-63c074a3-67d6-46cf-9018-790690d61473.png"> | <img width="563" alt="Screenshot 2022-10-07 at 11 03 48" src="https://user-images.githubusercontent.com/599268/194516671-f14242ce-807e-47d8-808c-c00b54de280d.png"> |

Maybe this can be a follow-up issue, but I would be very happy to improve our consistency here. 😌

Reply:
Follow-up thought: It does occur to me that, when we want one shade, we'll typically go for `bg-gray-100 dark:bg-gray-800`, but if we need two shades (e.g. "combined box with separator") maybe we want the second shade to be lighter both in light theme and dark theme?

I.e. we'll add the second shade `bg-gray-50`, but not `dark:bg-gray-900` -- instead, we'd shift the existing `dark:bg-gray-800` to `dark:bg-gray-700` and re-use `dark:bg-gray-800` as the "second shade"?

This would lead to inconsistencies between:
- `bg-gray-100 dark:bg-gray-800` alone (one background shade)
- `bg-gray-50 dark:bg-gray-800` combined with `bg-gray-100 dark:bg-gray-700`

Long story short, maybe we simply cannot have a 100% consistent mapping between light and dark shades, because we're using them differently in different contexts? 🤔

Reply:
I understand the concerns, especially since from a technical perspective this may look inconsistent or repetitive, but the colors used depend also on the layer they are used and the maturity of each component as they may evolve over time. This will get better as we create more reusable components and the product and components evolves. ➿ 

Like you [mentioned above](https://github.com/gitpod-io/gitpod/pull/13564#discussion_r989931274), because they are used in different contexts and sometimes different layers, it's impossible to have one mapping to rule them all to deliver a good UX. We need more grays!

Last but not least, the [screenshots above](https://github.com/gitpod-io/gitpod/pull/13564#discussion_r989866848) are a great example of :a: how we practice our values[[1](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#c2cdfc9a443644dd99e87031c375fbf3)][[2](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#30d4202757c74abc9821a1c79e0b1f75)][[3](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#2e0e717c03c04de882012e60ca348adc)][[4](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#a6c78fe159d547deb9426f30b7017aef)][[5](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#c8f962af695c4a43b4f39bdd3f5363a6)][[6](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#d766b4a2f4664dea9905ac29e65a5ccf)][[7](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#0c2e0d6c8a99404b97ccdc206b573d72)][[8](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#5c98845e99264744bf9c70ad710a1397)] and :b: what enabled us to make all the progress we've made so far in this small amount of time.